core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/AbstractUserStoreManager.java

```diff
@@ -12783,6 +12783,9 @@ private String getUserNameFromCurrentUserStore(String userID, UserStore userStor
         String userName = getFromUserNameCache(userID);
         if (StringUtils.isEmpty(userName)) {
             userName = doGetUserNameFromUserIDWithID(userID);
+            if (StringUtils.isEmpty(userName)) {
+                return null;
```

Comment:
Shall we add a log or a comment here? because this will occur due to the cache not updating? Otherwise this is kind of confusing how the username is in cache but not in the DB.